packages/instantsearch.js/src/lib/utils/hydrateSearchClient.ts

```diff
@@ -1,4 +1,6 @@
 // @ts-nocheck (types to be fixed during actual implementation)
+import qs from 'qs';
```

Comment:
not sure if the parsing is complex, `=` is encoded so we probably can do `split('&').reduce((acc, curr) => {const [key, val] = curr.split('='); acc[key] = val; return acc;}, {})` for the parse? 

that avoids leaking qs through more of the code base

Reply:
if we had the browser support, it could use URLSearchParams or FormData, but we can't for now, but that's no big loss I guess

Reply:
Well we would have to handle arrays as well as keys without `=` since that's what the other engine does.
I'll look into the other solution which to put the request in `initialResults`, at this point I think it's less trouble than writing our own parsing function which would have to be thoroughly tested

Reply:
```js
query.split('&').reduce((acc, curr) => {const [key, val] = curr.split('='); acc[key] = val ? decodeURIComponent(val) : '' ; return acc;}, {})
```

This seems to work fine actually, we don't need to parse the arrays

Reply:
Yes, the format is `encodeURI(param)=encodeURI(JSON.stringify(val))` I believe